ngraph/core/shape_inference/include/lstm_cell_shape_inference.hpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <rnn_cell_shape_inference.hpp>
+#include "shape_infer_utils.hpp"
+
+namespace ov {
+namespace op {
+namespace ShapeInferLSTM {
+template <class OpsType, class ShapeType>
+void lstm_shape_infer(const OpsType* op,
+                      const std::vector<ShapeType>& input_shapes,
+                      std::vector<ShapeType>& output_shapes,
+                      std::size_t gates_count) {
+    using DimType = typename std::iterator_traits<typename ShapeType::iterator>::value_type;
+    // If rank is dynamic, then output_shape is undefined
+    for (const auto& input : input_shapes) {
+        if (input.rank().is_dynamic()) {
+            ShapeInfer::default_work(output_shapes[0]);
+            ShapeInfer::default_work(output_shapes[1]);
+            return;
+        }
+    }
```

Comment:
This code misses a lot of the possibilities to calculate output shape for this operator. Please see the [specification](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/sequence/LSTMCell_1.md) -- there are a lot of places to deduce both `batch_size` and `hidden_size` dimension. Please extend it and add tests in type_prop tests
![image](https://user-images.githubusercontent.com/55839243/142296336-03df88b9-10ea-4333-a0c7-95749604d5f0.png)

Reply:
polish lstmcell in 4a44503e12d519ae51cc7989ffc24bce78c037a7